src/views/templates/details/TemplateNavPage.tsx

```diff
@@ -35,17 +34,17 @@ const TemplateNavPage: React.FC<TemplateNavPageProps> = ({
   });
   const pages = useVirtualMachineTabs();
 
+  const isTemplateEditable = useIsTemplateEditable(template);
+
   if (!loaded)
     return (
       <Bullseye>
         <Loading />
       </Bullseye>
     );
 
-  const isCommonTemplate = isCommonVMTemplate(template);
-
   return (
-    <SidebarEditorProvider isEditable={!isCommonTemplate}>
+    <SidebarEditorProvider isEditable={isTemplateEditable}>
```

Comment:
i think u can make the call of the hook inside SidebarEditorProvider , i will prevent rendering for other components.